assets/src/design-system/components/textArea/index.js

```diff
@@ -125,29 +134,40 @@ export const TextArea = forwardRef(
       showCount = false,
       maxLength,
       isIndeterminate = false,
+      onChange,
       ...props
     },
     ref
   ) => {
     const textAreaId = useMemo(() => id || uuidv4(), [id]);
     const textAreaRef = useRef(null);
-    const [focused, setFocused] = useState(false);
+    const [currentValue, setCurrentValue] = useState(value);
```

Comment:
IMO this component should be purely presentational like the rest of the form components in the design system (except the numeric input). If we want this to track state and submit when blurred, we should wrap this component in a container component that tracks that state

Reply:
Is the `NumericInput` used in the dashboard as well or could I move the additional non-presentational part to the editor?

Reply:
Yeah good question! I wanted to do the same for the `NumericInput` but I ended up making a compromise there due to time pressure since we wanted to get it in before y'all started work on the editor. I tried to keep the api as similar to the way y'all had used it already because it would be a nightmare to implement it without that submission functionality.

The `NumericInput` uses the presentational `Input` as a base component and then uses a hook to keep track of the 'submission' functionality: https://github.com/google/web-stories-wp/blob/main/assets/src/design-system/components/input/useNumericInput.js. This is similar to what I'm thinking we could do using this `TextArea` as a base component.

If I get the time to go back and change anything in there I'd try to rework it to make it so that we have a purely presentational `NumericInput` in the design system but have a wrapped component in the editor since that's where it's used 😄

Reply:
I love the idea of containerizing components out of the design system. I really think the design system should remain as presentational as possible. I think using the existing structures in the editor for everything non-presentational would give us a very clear line of separation. 

I think the reason this has gotten into a gray area is deadlines.